File: haskell/secret-handshake/src/SecretHandshake.hs
module SecretHandshake (handshake) where

import Data.Bits

construct :: Int -> [String]
construct n =
  map
    snd
    $ filter
      (testBit n . fst)
      $ zip [0..] actions
  where actions = ["wink", "double blink", "close your eyes", "jump"]

handshake :: Int -> [String]
handshake n
  | testBit n 4 = reverse $ construct n
  | otherwise   = construct n

-- vim:ts=2:sw=2:expandtab

File: python/secret-handshake/secret_handshake.py
CMDS = ('wink', 'double blink', 'close your eyes', 'jump')

def commands(number):
  number = int(number, 2)
  c = []
  for i, j in enumerate(CMDS):
    if number & 1 << i:
      c.append(j)
  if number & 1 << 4:
    c.reverse()
  return c


# vim:ts=2:sw=2:expandtab